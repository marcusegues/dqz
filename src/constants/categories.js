// @flow
import type { Categories } from '../types/basket';

export const CATEGORIES: Categories = {
  MEAT_AND_MEAT_PRODUCTS: 'MEAT_AND_MEAT_PRODUCTS',
  BUTTER_OR_CREAM: 'BUTTER_OR_CREAM',
  OILS_FATS_MARGARINE: 'OILS_FATS_MARGARINE',
  OTHER_FOODS: 'OTHER_FOODS',
  ALCOHOL_BELOW_18: 'ALCOHOL_BELOW_18',
  ALCOHOL_ABOVE_18: 'ALCOHOL_ABOVE_18',
  CIGARETTES_AND_CIGARS: 'CIGARETTES_AND_CIGARS',
  OTHER_TOBACCO: 'OTHER_TOBACCO',
  MEDICINE: 'MEDICINE',
  BOOKS: 'BOOKS',
  MAGAZINES: 'MAGAZINES',
  FLOWERS_AND_PLANTS: 'FLOWERS_AND_PLANTS',
  ANIMAL_FEED: 'ANIMAL_FEED',
  FERTILIZER: 'FERTILIZER',
  OTHER_GOODS: 'OTHER_GOODS',
};
