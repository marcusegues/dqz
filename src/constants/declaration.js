export const MAX_DECLARED_CHF = 20000;
