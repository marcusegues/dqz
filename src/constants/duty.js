export const dutyFreeAllowances = {
  MEAT_AND_MEAT_PRODUCTS: 1,
  BUTTER_OR_CREAM: 1,
  OILS_FATS_MARGARINE: 5,
  ALCOHOL_BELOW_18: 5,
  ALCOHOL_ABOVE_18: 1,
  CIGARETTES_AND_CIGARS: 250,
  OTHER_TOBACCO: 250,
};

export const MEAT_AND_MEAT_PRODUCTS_FIRST_DUTY_THRESHOLD = 10;
export const MEAT_AND_MEAT_PRODUCTS_DUTY_UNDER_THRESHOLD = 17;
export const MEAT_AND_MEAT_PRODUCTS_DUTY_OVER_THRESHOLD = 23;

export const BUTTER_OR_CREAM_DUTY = 16;
export const OILS_FATS_MARGARINE_DUTY = 2;
export const ALCOHOL_BELOW_18_DUTY = 2;
export const ALCOHOL_ABOVE_18_DUTY = 15;
export const CIGARETTES_AND_CIGARS_DUTY = 0.25;
export const OTHER_TOBACCO_DUTY = 0.1;
