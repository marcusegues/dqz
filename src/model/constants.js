// @flow

import type {
  CategoriesTypes,
  CategoryInfoType,
  RatesType,
} from './types/basketPeopleTypes';
import Immutable from 'immutable';
import {
  makeCategoryInfoRecord,
  makeDutyBracketRecord,
  makeRatesRecord,
} from './types/basketPeopleTypes';

export const IndividualAllowanceAdult: number = 300;
export const IndividualAllowanceMinor: number = 300;

export const CATEGORIES: Set<CategoriesTypes> = new Set([
  'Meat',
  'Butter',
  'Oils',
  'OtherFood',
  'AlcSoft',
  'AlcHard',
  'Cigarettes',
  'Tobacco',
  'Meds',
  'Books',
  'Magazines',
  'Flowers',
  'AnimalFeed',
  'Fertilizer',
  'Other',
]);

export const CategoriesArray = Array.from(CATEGORIES.values());

export const CategoriesRates: RatesType = Immutable.Map([
  [
    'Meat',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List([
        makeDutyBracketRecord({ threshold: 10, fee: 17 }),
        makeDutyBracketRecord({ threshold: Infinity, fee: 23 }),
      ]),
      dutyAllowance: 1,
      adultsOnly: false,
    }),
  ],
  [
    'Butter',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List([
        makeDutyBracketRecord({ threshold: Infinity, fee: 16 }),
      ]),
      dutyAllowance: 1,
      adultsOnly: false,
    }),
  ],
  [
    'Oils',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List([
        makeDutyBracketRecord({ threshold: Infinity, fee: 2 }),
      ]),
      dutyAllowance: 5,
      adultsOnly: false,
    }),
  ],
  [
    'OtherFood',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
  [
    'AlcSoft',
    makeRatesRecord({
      vat: 0.08,
      duty: Immutable.List([
        makeDutyBracketRecord({ threshold: Infinity, fee: 2 }),
      ]),
      dutyAllowance: 5,
      adultsOnly: true,
    }),
  ],
  [
    'AlcHard',
    makeRatesRecord({
      vat: 0.08,
      duty: Immutable.List([
        makeDutyBracketRecord({ threshold: Infinity, fee: 15 }),
      ]),
      dutyAllowance: 1,
      adultsOnly: true,
    }),
  ],
  [
    'Cigarettes',
    makeRatesRecord({
      vat: 0.08,
      duty: Immutable.List([
        makeDutyBracketRecord({ threshold: Infinity, fee: 0.25 }),
      ]),
      dutyAllowance: 250,
      adultsOnly: false,
    }),
  ],
  [
    'Tobacco',
    makeRatesRecord({
      vat: 0.08,
      duty: Immutable.List([
        makeDutyBracketRecord({ threshold: Infinity, fee: 0.1 }),
      ]),
      dutyAllowance: 250,
      adultsOnly: true,
    }),
  ],
  [
    'Meds',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
  [
    'Books',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
  [
    'Magazines',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
  [
    'Flowers',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
  [
    'AnimalFeed',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
  [
    'Fertilizer',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
  [
    'Other',
    makeRatesRecord({
      vat: 0.025,
      duty: Immutable.List(),
      dutyAllowance: Infinity,
      adultsOnly: false,
    }),
  ],
]);

export const CategoriesInfo: CategoryInfoType = Immutable.Map([
  [
    'Meat',
    makeCategoryInfoRecord({
      name: 'Fleisch und Fleischzubereitung',
      unit: 'kg',
      icon: 'meat',
    }),
  ],
  [
    'Butter',
    makeCategoryInfoRecord({
      name: 'Butter und Rahm',
      unit: 'kg/Liter',
      icon: 'butter',
    }),
  ],
  [
    'Oils',
    makeCategoryInfoRecord({
      name: 'Öle, Fette, Margarine zu Speisezwecken',
      unit: 'kg/Liter',
      icon: 'oil',
    }),
  ],
  [
    'OtherFood',
    makeCategoryInfoRecord({
      name: 'Sonstige Lebensmittel & Alkoholfreie Getranke',
      unit: 'kg/Liter',
      icon: 'oil',
    }),
  ],
  [
    'Other',
    makeCategoryInfoRecord({
      name: 'Sonstige Lebensmittel & Alkoholfreie Getranke',
      unit: 'kg/Liter',
      icon: 'oil',
    }),
  ],
  [
    'AlcSoft',
    makeCategoryInfoRecord({
      name: 'Alkoholische Getränke, Alkoholgehalt bis 18% Vol.',
      unit: 'Liter',
      icon: 'beer',
    }),
  ],
  [
    'AlcHard',
    makeCategoryInfoRecord({
      name: 'Alkoholische Getränke, Alkoholgehalt über 18% Vol.',
      unit: 'Liter',
      icon: 'vodka',
    }),
  ],
  [
    'Cigarettes',
    makeCategoryInfoRecord({
      name: 'Zigaretten/Zigarren',
      unit: 'Stück',
      icon: 'cigarette',
    }),
  ],
  [
    'Tobacco',
    makeCategoryInfoRecord({
      name: 'Andere Tabakfabrikate',
      unit: 'Gramm',
      icon: 'cigarette',
    }),
  ],
  [
    'Meds',
    makeCategoryInfoRecord({
      name: 'Medikamente',
      unit: '',
      icon: 'cigarette',
    }),
  ],
  [
    'Books',
    makeCategoryInfoRecord({
      name: 'Bücher',
      unit: '',
      icon: 'cigarette',
    }),
  ],
  [
    'Magazines',
    makeCategoryInfoRecord({
      name: 'Zeitschriften',
      unit: '',
      icon: 'cigarette',
    }),
  ],
  [
    'Flowers',
    makeCategoryInfoRecord({
      name: 'Schnittblumen & Pflanzen',
      unit: '',
      icon: 'cigarette',
    }),
  ],
  [
    'AnimalFeed',
    makeCategoryInfoRecord({
      name: 'Tierfutter',
      unit: '',
      icon: 'cigarette',
    }),
  ],
  [
    'Fertilizer',
    makeCategoryInfoRecord({
      name: 'Dünger',
      unit: '',
      icon: 'cigarette',
    }),
  ],
]);
