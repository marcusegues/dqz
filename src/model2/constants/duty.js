export const dutyFreeAllowances = {
  MEAT_AND_MEAT_PRODUCTS: 1,
  BUTTER_OR_CREAM: 1,
  OILS_FATS_MARGARINE: 5,
  ALCOHOL_BELOW_18: 5,
  ALCOHOL_ABOVE_18: 1,
  CIGARETTES_AND_CIGARS: 250,
  OTHER_TOBACCO: 250,
};

export const dutyRates = {
  MEAT_AND_MEAT_PRODUCTS_FIRST_DUTY_THRESHOLD: 10,
  MEAT_AND_MEAT_PRODUCTS_DUTY_UNDER_THRESHOLD: 17,
  MEAT_AND_MEAT_PRODUCTS_DUTY_OVER_THRESHOLD: 23,
  BUTTER_OR_CREAM_DUTY: 16,
  OILS_FATS_MARGARINE_DUTY: 2,
  ALCOHOL_BELOW_18_DUTY: 2,
  ALCOHOL_ABOVE_18_DUTY: 15,
  CIGARETTES_AND_CIGARS_DUTY: 0.25,
  OTHER_TOBACCO_DUTY: 0.1,
};
