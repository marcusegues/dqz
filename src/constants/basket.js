// @flow
import { CATEGORIES_LIST } from './categories';
import type { Basket } from '../types/basket';

let emptyBasket: Basket = {};
CATEGORIES_LIST.forEach(category => {
  emptyBasket[category] = {
    quantity: 0,
    duty: 0,
    values: [],
  };
});

export { emptyBasket };
