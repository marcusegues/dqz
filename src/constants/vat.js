// @flow
import { CATEGORIES } from './categories';
import type { VatRates } from '../types/basket';

// any changes to the vat rates go here
export const VAT_RATES: VatRates = {
  [CATEGORIES.MEAT_AND_MEAT_PRODUCTS]: 0.025,
  [CATEGORIES.BUTTER_OR_CREAM]: 0.025,
  [CATEGORIES.OILS_FATS_MARGARINE]: 0.025,
  [CATEGORIES.OTHER_FOODS]: 0.025,
  [CATEGORIES.ALCOHOL_BELOW_18]: 0.08,
  [CATEGORIES.ALCOHOL_ABOVE_18]: 0.08,
  [CATEGORIES.CIGARETTES_AND_CIGARS]: 0.08,
  [CATEGORIES.OTHER_TOBACCO]: 0.08,
  [CATEGORIES.MEDICINE]: 0.025,
  [CATEGORIES.BOOKS]: 0.025,
  [CATEGORIES.MAGAZINES]: 0.025,
  [CATEGORIES.FLOWERS_AND_PLANTS]: 0.025,
  [CATEGORIES.ANIMAL_FEED]: 0.025,
  [CATEGORIES.FERTILIZER]: 0.025,
  [CATEGORIES.OTHER_GOODS]: 0.08,
};
