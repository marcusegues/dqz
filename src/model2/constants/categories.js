// @flow
import type {
  CategoryName,
  Categories,
  CategoriesList,
  CategoriesInfoType,
} from '../types/basket';
import { makeCategoryInfoRecord } from '../types/basket';
import { List as ImmutableListType } from 'immutable';
import Immutable from 'immutable';

// CATEGORIES and CATEGORIES_LIST are the two base objects containing
// all the categories. If a new category needs to be added, it is added to
// both of these objects

export const CATEGORIES: Categories = {
  MEAT_AND_MEAT_PRODUCTS: 'MEAT_AND_MEAT_PRODUCTS',
  BUTTER_OR_CREAM: 'BUTTER_OR_CREAM',
  OILS_FATS_MARGARINE: 'OILS_FATS_MARGARINE',
  OTHER_FOODS: 'OTHER_FOODS',
  ALCOHOL_BELOW_18: 'ALCOHOL_BELOW_18',
  ALCOHOL_ABOVE_18: 'ALCOHOL_ABOVE_18',
  CIGARETTES_AND_CIGARS: 'CIGARETTES_AND_CIGARS',
  OTHER_TOBACCO: 'OTHER_TOBACCO',
  MEDICINE: 'MEDICINE',
  BOOKS: 'BOOKS',
  MAGAZINES: 'MAGAZINES',
  FLOWERS_AND_PLANTS: 'FLOWERS_AND_PLANTS',
  ANIMAL_FEED: 'ANIMAL_FEED',
  FERTILIZER: 'FERTILIZER',
  OTHER_GOODS: 'OTHER_GOODS',
};

export const CATEGORIES_LIST: CategoriesList = Immutable.List([
  CATEGORIES.MEAT_AND_MEAT_PRODUCTS,
  CATEGORIES.BUTTER_OR_CREAM,
  CATEGORIES.OILS_FATS_MARGARINE,
  CATEGORIES.OTHER_FOODS,
  CATEGORIES.ALCOHOL_BELOW_18,
  CATEGORIES.ALCOHOL_ABOVE_18,
  CATEGORIES.CIGARETTES_AND_CIGARS,
  CATEGORIES.OTHER_TOBACCO,
  CATEGORIES.MEDICINE,
  CATEGORIES.BOOKS,
  CATEGORIES.MAGAZINES,
  CATEGORIES.FLOWERS_AND_PLANTS,
  CATEGORIES.ANIMAL_FEED,
  CATEGORIES.FERTILIZER,
  CATEGORIES.OTHER_GOODS,
]);

// null unit means it's zollfrei
export const CATEGORIES_INFO: CategoriesInfoType = Immutable.Map([
  [
    CATEGORIES.MEAT_AND_MEAT_PRODUCTS,
    makeCategoryInfoRecord({
      name: 'Fleisch und Fleischzubereitung',
      unit: 'kg',
      icon: 'meat',
    }),
  ],
  [
    CATEGORIES.BUTTER_OR_CREAM,
    makeCategoryInfoRecord({
      name: 'Butter und Rahm',
      unit: 'kg/Liter',
      icon: 'butter',
    }),
  ],
  [
    CATEGORIES.OILS_FATS_MARGARINE,
    makeCategoryInfoRecord({
      name: 'Öle, Fette, Margarine zu Speisezwecken',
      unit: 'kg/Liter',
      icon: 'oil',
    }),
  ],
  [
    CATEGORIES.OTHER_FOODS,
    makeCategoryInfoRecord({
      name: 'Sonstige Lebensmittel & Alkoholfreie Getranke',
      unit: 'kg/Liter',
      icon: 'oil',
    }),
  ],
  [
    CATEGORIES.ALCOHOL_BELOW_18,
    makeCategoryInfoRecord({
      name: 'Alkoholische Getränke, Alkoholgehalt bis 18% Vol.',
      unit: 'Liter',
      icon: 'beer',
    }),
  ],
  [
    CATEGORIES.ALCOHOL_ABOVE_18,
    makeCategoryInfoRecord({
      name: 'Alkoholische Getränke, Alkoholgehalt über 18% Vol.',
      unit: 'Liter',
      icon: 'vodka',
    }),
  ],
  [
    CATEGORIES.CIGARETTES_AND_CIGARS,
    makeCategoryInfoRecord({
      name: 'Zigaretten/Zigarren',
      unit: 'Stück',
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.OTHER_TOBACCO,
    makeCategoryInfoRecord({
      name: 'Andere Tabakfabrikate',
      unit: 'Gramm',
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.MEDICINE,
    makeCategoryInfoRecord({
      name: 'Medikamente',
      unit: null,
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.BOOKS,
    makeCategoryInfoRecord({
      name: 'Bucher',
      unit: null,
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.MAGAZINES,
    makeCategoryInfoRecord({
      name: 'Zeitschriften',
      unit: null,
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.FLOWERS_AND_PLANTS,
    makeCategoryInfoRecord({
      name: 'Schnittblumen & Pflanzen',
      unit: null,
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.ANIMAL_FEED,
    makeCategoryInfoRecord({
      name: 'Tierfutter',
      unit: null,
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.FERTILIZER,
    makeCategoryInfoRecord({
      name: 'Dunger',
      unit: null,
      icon: 'cigarette',
    }),
  ],
  [
    CATEGORIES.OTHER_GOODS,
    makeCategoryInfoRecord({
      name: 'Sonstige Waren',
      unit: null,
      icon: 'cigarette',
    }),
  ],
]);
