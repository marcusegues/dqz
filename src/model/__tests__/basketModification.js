// @flow
import type { Basket } from '../types/basketPeopleTypes';
import {
  addAdult,
  addAmount,
  addLargeAmount,
  addMinor,
  emptyBasket,
  getAdultPeople,
  getAmounts,
  getLargeAmounts,
  getMinorPeople,
  getQuantity,
  initPeople,
  resetAmounts,
  resetLargeAmounts,
  setAdultPeople,
  setMinorPeople,
  setQuantities,
  setQuantity,
  subtractAdult,
  subtractMinor,
} from '../configurationApi';
import { CategoriesArray } from '../constants';
import Immutable from 'immutable';

const basket1: Basket = emptyBasket;

const quantityBasket1: Basket = setQuantity(basket1, 'Butter', 1234);
const quantityBasket2: Basket = setQuantities(basket1, [
  { category: 'Meat', quantity: 123 },
  { category: 'Butter', quantity: 321 },
]);
const quantityBasket3: Basket = basket1;

const amountBasket1: Basket = addAmount(basket1, 'Butter', 12.34);
const amountBasket2: Basket = addAmount(amountBasket1, 'Butter', 34.56);

const largeAmountBasket1: Basket = addLargeAmount(basket1, 'Tobacco', 1234);
const largeAmountBasket2: Basket = addLargeAmount(
  largeAmountBasket1,
  'Tobacco',
  1234
);

const twoAdultsNoMinor = addAdult(initPeople);
const noAdultsNoMinor = subtractAdult(initPeople);
const oneAdultOneMinor = addMinor(initPeople);

describe('The basket / quantites: ', () => {
  test('defaults to 0 quantity', () => {
    CategoriesArray.forEach(c => expect(getQuantity(emptyBasket, c)).toBe(0));
  });
  test('has individually set quantities: ', () => {
    expect(getQuantity(quantityBasket1, 'Butter')).toBe(1234);
  });
  test('leaves other quantities untouched: ', () => {
    expect(getQuantity(quantityBasket1, 'Meat')).toBe(0);
  });
  test('has sync set quantities: ', () => {
    expect(getQuantity(quantityBasket2, 'Butter')).toBe(321);
    expect(getQuantity(quantityBasket2, 'Meat')).toBe(123);
  });
  test('is immutable', () => {
    CategoriesArray.forEach(c =>
      expect(getQuantity(quantityBasket3, c)).toBe(0)
    );
  });
});

describe('The basket / amounts: ', () => {
  test('defaults to empty set of amounts', () => {
    CategoriesArray.forEach(c =>
      expect(getAmounts(emptyBasket, c)).toBe(Immutable.List())
    );
  });
  test('adds amounts: ', () => {
    expect(getAmounts(amountBasket1, 'Butter').toString()).toBe(
      Immutable.List([12.34]).toString()
    );
  });
  test('adds another amount: ', () => {
    expect(getAmounts(amountBasket2, 'Butter').toString()).toBe(
      Immutable.List([12.34, 34.56]).toString()
    );
  });
  test('leaves other amounts untouched: ', () => {
    expect(getAmounts(amountBasket1, 'Meat')).toBe(Immutable.List());
  });
  test('resets amounts: ', () => {
    const resetted: Basket = resetAmounts(amountBasket1, 'Butter');
    expect(getAmounts(resetted, 'Butter')).toBe(Immutable.List());
  });
});

describe('The basket / large amounts: ', () => {
  test('defaults to empty set of amounts', () => {
    CategoriesArray.forEach(c =>
      expect(getLargeAmounts(emptyBasket, c)).toBe(Immutable.List())
    );
  });
  test('adds large amounts: ', () => {
    expect(getLargeAmounts(largeAmountBasket1, 'Tobacco').toString()).toBe(
      Immutable.List([1234]).toString()
    );
  });
  test('adds another large amount: ', () => {
    expect(getLargeAmounts(largeAmountBasket2, 'Tobacco').toString()).toBe(
      Immutable.List([1234, 1234]).toString()
    );
  });
  test('leaves other large amounts untouched: ', () => {
    expect(getLargeAmounts(amountBasket1, 'Meat')).toBe(Immutable.List());
  });
  test('resets large amounts: ', () => {
    const resetted: Basket = resetLargeAmounts(amountBasket1, 'Tobacco');
    expect(getLargeAmounts(resetted, 'Tobacco')).toBe(Immutable.List());
  });
});

describe('Party configuration', () => {
  test('inits with 1 adult and no minors', () => {
    expect(getAdultPeople(initPeople)).toBe(1);
    expect(getMinorPeople(initPeople)).toBe(0);
  });
  test('adds adults', () => {
    expect(getAdultPeople(twoAdultsNoMinor)).toBe(2);
  });
  test('subtracts adults', () => {
    expect(getAdultPeople(noAdultsNoMinor)).toBe(0);
  });
  test('subtracts adults (non-negative', () => {
    expect(getAdultPeople(subtractAdult(noAdultsNoMinor))).toBe(0);
  });

  test('adds minor', () => {
    expect(getMinorPeople(oneAdultOneMinor)).toBe(1);
  });
  test('subtracts minor', () => {
    expect(getMinorPeople(subtractMinor(oneAdultOneMinor))).toBe(0);
  });
  test('subtracts minor (non-negative)', () => {
    expect(getMinorPeople(subtractMinor(noAdultsNoMinor))).toBe(0);
  });
  test('set 11 adults', () => {
    expect(getAdultPeople(setAdultPeople(initPeople, 11))).toBe(11);
  });
  test('set 11 adults on init -> still no minors', () => {
    expect(getMinorPeople(setAdultPeople(initPeople, 11))).toBe(0);
  });
  test('set 11 minor', () => {
    expect(getMinorPeople(setMinorPeople(initPeople, 11))).toBe(11);
  });
  test('set 11 adults on init -> still no minors', () => {
    expect(getAdultPeople(setMinorPeople(initPeople, 11))).toBe(1);
  });
  test('set -11 adults', () => {
    expect(getAdultPeople(setAdultPeople(initPeople, -11))).toBe(0);
  });
  test('set -11 minor', () => {
    expect(getMinorPeople(setMinorPeople(initPeople, -11))).toBe(0);
  });
});
