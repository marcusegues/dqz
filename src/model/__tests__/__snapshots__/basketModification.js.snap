// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The basket / quantites:  can reset the quantites of a category 1`] = `
Immutable.Map {
  "Butter": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [
        23,
        23,
        23,
      ],
    },
  },
  "Oils": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Cigarettes": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "AlcSoft": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Fertilizer": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "AlcHard": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Meds": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Books": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Meat": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "OtherFood": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Other": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Tobacco": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [
        3219,
        3219,
        3219,
      ],
    },
  },
  "Flowers": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "AnimalFeed": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Magazines": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
}
`;

exports[`The basket / quantites:  can reset the quantites of multiple categorie 1`] = `
Immutable.Map {
  "Butter": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Oils": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Cigarettes": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "AlcSoft": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Fertilizer": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "AlcHard": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Meds": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Books": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Meat": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "OtherFood": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Other": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Tobacco": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [
        3219,
        3219,
        3219,
      ],
    },
  },
  "Flowers": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "AnimalFeed": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
  "Magazines": Immutable.Record {
    "volume": Immutable.Record {
      "quantities": Immutable.List [],
    },
  },
}
`;
